attestation/aws-iid/aws-iid.go

```diff
@@ -182,12 +182,13 @@ func (a *Attestor) Verify() error {
 	a.PublicKey = string(pem)
 
 	if err != nil {
-		return fmt.Errorf("failed to encode public key: %v", err)
+		return fmt.Errorf("failed to encode public key: %w", err)
 	}
 
 	err = rsa.VerifyPKCS1v15(pubKey, crypto.SHA256, docHash[:], sigBytes)
 	if err != nil {
-		log.Debugf("(attestation/aws-iid) failed to verify signature: %v", err)
+		err := fmt.Errorf("(attestation/aws-iid) failed to verify signature: %w", err)
+		log.Debug(err)
```

Comment:
Would this work?

```log.Errorf("(attestation/aws-iid) failed to verify signature: %w", err)```

Reply:
So I've just done a bit more digging on this and it seems I slightly misconveyed things.

`go-witness` has [package log](https://github.com/in-toto/go-witness/blob/main/log/log.go) which is an interface to allow different logging libraries to be plugged in. 

We use `logrus` in https://github.com/in-toto/witness and _I believe_ `log.Errorf` in itself does not support error wrapping with the use of `%w`. This is why I decided to use `fmt.Errorf` separately. Instead, it allows for the use of `WithError` and from what I can tell, expects you to wrap errors that way.

We could work round this by expanding the package so the interface implements `WithError`, but that would leave those wanting to leverage the logging inside the `go-witness` library needing to implement this themselves. This (I think) also holds true if we wish to more deeply use more of the `logrus` library's features along the line.

The way I see it (and @jkjell would love to know your opinion along with others), we have a few options:
a) expand the interface whenever we need so we can use logrus features. This seems like a bad idea to me as if those utilising the library are using other loggers, it would be a pain.
b) handle this (and other situations like it) with some utility function inside the `log` package. This actually seems the right way to me, I can neaten all this up right now by adding this to the package. This could possibly lead to situations where we are handing things back to the logger in an incorrect form though? Not sure if that is a problem.
c) Don't do any logging inside the library. I think this would be troublesome for Debug logs, I know for a fact that logging implemented inside `cosign` libraries has come in handy in the past when I have been doing projects.

Any thoughts welcome 😄

Reply:
@mikhailswift what are your opinions here?

Reply:
I am heavily leaning toward b, and I think @ChaosInTheCRD 's reasoning is sound. 

The reasoning for implementing the log interface within the go-witness library was to keep it generic enough for users of the library to use their logging library of choice -- I didn't want the library to be too opinionated. Libraries logging to stdout or stderr unexpectedly creates a frustrating experience for library users. c kind of hints at this, there's a balance where you don't want a library to log on you, but debug logging can be extraordinarily useful.

Reply:
great stuff @ChaosInTheCRD! 

I also like the sounds of B, try to reduce the amount of pain our users feel

Reply:
So I figured what we could do is check for an error in the relevant `f` functions (e.g., `Debugf`, `Warnf`) for arguments of type `error`, and handle that inside the wrapper function. I have it in a new branch (to get rid of the now unnecessary refactoring) [here](https://github.com/ChaosInTheCRD/go-witness/blob/refactoring-error-wrapping/log/log.go#L69-L76).

Thoughts would be appreciated. If this works I can go ahead and close this PR and I'll raise a PR for the other branch 😄 .